server.go

```diff
@@ -98,13 +98,30 @@ func runMain() error {
 		}
 	}
 
+	mainControlCfg := controllers.MainControllerConfig{
```

Comment:
Seems to me that `cfg` can be passed directly instead of copying over certain values from `cfg` to `mainControlCfg`

Reply:
This is the challenge with this issue. The config type cannot be exported from the main package to the controllers package.